rdrf/rdrf/admin.py

```diff
@@ -178,6 +184,89 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('Yes')),
+            ('False', _('No')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_profiles = [profile.id for profile in queryset if profile.activation_key_expired()]
+
+        if self.value() == 'True':
+            return queryset.filter(id__in=expired_profiles)
+        elif self.value() == 'False':
+            return queryset.exclude(id__in=expired_profiles)
+        else:
+            return queryset
+
+
+class CustomRegistrationProfileAdmin(RegistrationAdmin):
+    list_display = ('user', 'custom_activation_key_expired', 'is_activated')
+    list_filter = ['activated', ActivationKeyExpirationListFilter]
+
+    @admin.display(description="Activation key expired")
+    def custom_activation_key_expired(self, obj):
+        max_expiry_days = datetime.timedelta(days=settings.ACCOUNT_ACTIVATION_DAYS)
+        activation_date = obj.user.date_activated
+        expiration_date = (activation_date or obj.user.date_joined) + max_expiry_days
+
+        return obj.activated or expiration_date <= datetime.datetime.now()
+
+    @admin.display(description="Activated")
+    def is_activated(self, obj):
+        return obj.activated
+
+    def activate_user(self, activation_key):
+        sha256_re = re.compile('^[a-f0-9]{40,64}$')
+
+        def activate(user_profile):
+            user = user_profile.user
+            user.is_active = True
+            user_profile.activated = True
+
+            with transaction.atomic():
+                user.save()
+                user_profile.save()
+
+            return user
+
+        if sha256_re.search(activation_key):
+            profile = RegistrationProfile.objects.filter(activation_key=activation_key)
+            if not profile.exists():
+                return False, False
+
+            profile = profile.first()
+            if not self.custom_activation_key_expired(profile):
```

Comment:
Do you need to check if `profile.activated` similar to how it's checked on the `RegistrationManager` from the `registration` package?